azure-mgmt-keyvault/src/main/java/com/microsoft/azure/management/keyvault/Vault.java

```diff
@@ -87,6 +87,24 @@
      * retrieve secrets from the key vault.
      */
     boolean enabledForTemplateDeployment();
+    
+    /**
+     * @return whether soft delete is enabled for this key vault.
+     */
+    boolean softDeleteEnabled();
```

Comment:
@tiffanyachen in this case you need to be consistent with the rest of the Java SDK naming patterns for this service and not other languages. There's a good reason and lots of review discussions before we chose the naming we use today. So unless you want to change all the other method names and introduce your own new naming pattern, then work around the backward compatibility breaks resulted from these change, then my recommendation is to stay with @anuchandy feedback.

Reply:
@anuchandy @milismsft In the case of the first three fields, enableForDeployment, enabledForDiskEncryption, and enabledForTemplateDeployment, they are named this way because the user is enabling the vault to be accesed **FOR** arm deployments, disk encryption, and template deployment.  In the case of softDeleteEnabled and purgeProtectionEnabled the user is enabling the features soft delete and purge protection on the vault.

Reply:
@schaabs I don't understand how is soft delete any different then the others... It looks like it is required for instance when setting up SQL TDE (see customer report https://github.com/Azure/azure-libraries-for-net/issues/315).
If "soft delete" and "purge protection" are simple vault properties it might be worth removing the whole "enabled" thing from the name, i.e "bool softDelete()" and "bool purgeProtection()".